src/main/kotlin/dev/supergrecko/kllvm/core/typedefs/Builder.kt

```diff
@@ -2,38 +2,77 @@ package dev.supergrecko.kllvm.core.typedefs
 
 import dev.supergrecko.kllvm.contracts.Disposable
 import dev.supergrecko.kllvm.contracts.Validatable
+import dev.supergrecko.kllvm.core.values.InstructionValue
 import org.bytedeco.llvm.LLVM.LLVMBuilderRef
 import org.bytedeco.llvm.global.LLVM
 
 public class Builder internal constructor(internal val llvmBuilder: LLVMBuilderRef) : AutoCloseable, Validatable, Disposable {
     public override var valid: Boolean = true
 
-    override fun dispose() {
-        require(valid) { "This builder has already been disposed." }
+    //region Core::Builders
+    public fun getUnderlyingRef(): LLVMBuilderRef {
+        return llvmBuilder
+    }
 
-        valid = false
+    public fun buildRetVoid(): Value {
+        return Value(LLVM.LLVMBuildRetVoid(llvmBuilder))
+    }
 
-        LLVM.LLVMDisposeBuilder(llvmBuilder)
+    /**
+     * LLVMPositionBuilder
+     */
+    public fun positionBefore(instruction: InstructionValue): Unit {
+        // TODO: Test
+        LLVM.LLVMPositionBuilderBefore(getUnderlyingRef(), instruction.llvmValue)
     }
 
-    override fun close() = dispose()
+    /**
+     * LLVMPositionBuilderAtEnd
+     */
+    public fun positionAtEnd(basicBlock: BasicBlock): Unit {
+        LLVM.LLVMPositionBuilderAtEnd(getUnderlyingRef(), basicBlock.llvmBlock)
+    }
 
-    fun positionAtEnd(basicBlock: BasicBlock) {
-        LLVM.LLVMPositionBuilderAtEnd(llvmBuilder, basicBlock.llvmBlock)
+    /**
+     * LLVMGetInsertBlock
+     */
+    public fun getInsertBlock(): BasicBlock? {
+        val ref = LLVM.LLVMGetInsertBlock(getUnderlyingRef()) ?: return null
+        return BasicBlock(ref)
     }
 
-    fun getUnderlyingRef(): LLVMBuilderRef {
-        return llvmBuilder
+    /**
+     * LLVMClearInsertionPosition
+     */
+    public fun clearInsertPosition(): Unit = LLVM.LLVMClearInsertionPosition(getUnderlyingRef())
+
+    /**
+     * LLVMInsertIntoBuilderWithName
+     */
+    public fun insert(instruction: InstructionValue, name: String?): Unit {
+        // TODO: Test
+        LLVM.LLVMInsertIntoBuilderWithName(getUnderlyingRef(), instruction.getUnderlyingReference(), name)
     }
 
-    fun buildRetVoid(): Value {
-        return Value(LLVM.LLVMBuildRetVoid(llvmBuilder))
+    //endregion Core::Builders
+
+    override fun dispose() {
+        require(valid) { "This builder has already been disposed." }
+
+        valid = false
+
+        LLVM.LLVMDisposeBuilder(llvmBuilder)
     }
 
+    override fun close() = dispose()
+
     companion object {
+        //region Core::Builders
         @JvmStatic
-        fun create(ctx: Context): Builder {
-            return Builder(LLVM.LLVMCreateBuilderInContext(ctx.llvmCtx))
+        fun create(ctx: Context? = null): Builder {
```

Comment:
While this may work, I think we should default it to the global context like LLVM does for consistency (see https://llvm.org/doxygen/IR_2Core_8cpp_source.html#l02991)

Reply:
Updated